daft/expressions/expressions.py

```diff
@@ -3109,7 +3116,9 @@ def extract(self, pattern: str | Expression, index: int = 0) -> Expression:
             [extract_all](https://www.getdaft.io/projects/docs/en/stable/api/expressions/#daft.expressions.expressions.ExpressionStringNamespace.extract_all)
         """
         pattern_expr = Expression._to_expression(pattern)
-        return Expression._from_pyexpr(native.utf8_extract(self._expr, pattern_expr._expr, index))
+        idx = Expression._to_expression(index)
+        f = native.get_function_from_registry("regexp_extract")
+        return Expression._from_pyexpr(f(self._expr, pattern_expr._expr, idx._expr))
```

Comment:
We could cut out boilerplate with something like,

```python
def __apply(self, function: str, args: list[object] = []) -> Expression:
      func = native.get_function_from_registry(function)
      argv = [ Expression._to_expression(arg)._expr for arg in args ]
      return Expression._from_pyexpr(func(self._expr, *argv)
```

Then most functions become one-liners like,

```python
def extract(self, pattern: str | Expression, index: int = 0) -> Expression:
    return self.__apply("extract", pattern, index)
```

This would at least cut things in half and reduce all the repetition here, where all bridging translation logic is consolidated. It is more helpful for functions with multiple arguments, but still cuts the unary function translations in half.

Reply:
I have this addressed in a followup PR!

Reply:
Do you have a link for that PR?

Reply:
https://github.com/Eventual-Inc/Daft/pull/4340. 

It doesn't rewrite all of the exprs from this PR to use that.

Once both are in, I can followup and rewrite all of these to use that pattern.